bin/interpreter.sh

```diff
@@ -205,7 +205,7 @@ fi
 
 addJarInDirForIntp "${LOCAL_INTERPRETER_REPO}"
 
-if [[ ! -z "$ZEPPELIN_IMPERSONATE_USER" && "${INTERPRETER_ID}" != "spark" ]]; then
+if [[ ! -z "$ZEPPELIN_IMPERSONATE_USER" && "$ZEPPELIN_IMPERSONATE_SPARK_PROXY_USER" == "false" ]]; then
```

Comment:
hmm, I'm actually not sure this is the way to run spark job as the user

Reply:
In case of ZEPPELIN_IMPERSONATE_SPARK_PROXY_USER = true, spark interpreter should run as zeppelin user with --proxy-user=[impersonated user].
In case of ZEPPELIN_IMPERSONATE_SPARK_PROXY_USER = false, spark interpreter should run as impersonated user with no --proxy-user option.

But Currently, 
In case of ZEPPELIN_IMPERSONATE_SPARK_PROXY_USER = false, spark interpreter runs as zeppelin user with no --proxy-user option.
This is due to "${INTERPRETER_ID}" != "spark" of if condition.

If condition should include ZEPPELIN_IMPERSONATE_SPARK_PROXY_USER = false instead of "${INTERPRETER_ID}" != "spark".

Reply:
@zjffdu Please could you review this?
This logic was recently modified by your PR #2705.

Reply:
@iijima-satoshi Have you verified other interpreters like shell ?

Reply:
@zjffdu I fixed this line. Please see https://github.com/apache/zeppelin/pull/2916/files
I have verified test cases including other interpreters.